src/main/java/org/commonjava/cdi/util/weft/PoolWeftExecutorService.java

```diff
@@ -131,79 +135,111 @@ public boolean awaitTermination( long l, TimeUnit timeUnit )
         return delegate.awaitTermination( l, timeUnit );
     }
 
+    private void verifyLoad()
+    {
+        if ( loadSensitive && !isHealthy() )
+        {
+            throw new PoolOverloadException( getName(), getLoadFactor(), getCurrentLoad(), getThreadCount() );
```

Comment:
One question here: how will we control this exception in indy? Should we catch this exception in indy's calling points of all WeftExcecutorService and handle them separately?

Reply:
What about making this exception extend Executioner option? Is that in the method signature?

Reply:
I'm resolving this, so I can merge it and try to modify things slightly so it's a little easier to integrate into Indy. We may end up changing this later, but I want to get something built and start getting some feedback / mileage on it.